packages/renderer/src/stores/navigation/kubernetes/navigation-registry-k8s-pods.svelte.spec.ts

```diff
@@ -0,0 +1,61 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { KubernetesObject } from '@kubernetes/client-node';
+import { beforeEach, expect, test, vi } from 'vitest';
+
+import { createNavigationKubernetesDeploymentsEntry } from './navigation-registry-k8s-deployments.svelte';
+
+beforeEach(() => {
+  vi.resetAllMocks();
+  Object.defineProperty(window, 'kubernetesRegisterGetCurrentContextResources', {
+    value: kubernetesRegisterGetCurrentContextResourcesMock,
+  });
+});
+
+const kubernetesRegisterGetCurrentContextResourcesMock = vi.fn();
```

Comment:
We should not define mock like this, we should use `vi.mocked(window.kubernetesRegisterGetCurrentContextResources)` to access it, defining mock like this lose all the typing of the mocked function

Reply:
Updated to use vi.mocked in new commit (expect ResizeObserver, which I couldn't figure out).

Side note: I agree with the improved standards, but now we'll have 7 'copy/paste' Kubernetes resource files, and this one that's better but unique. Need to make sure we follow-up and make the others match.